modules/photon/system-part1/src/newlib.cpp

```diff
@@ -1 +1,4 @@
 #include "../../../../hal/src/stm32/newlib.cpp"
+
+// Defining this in hal/src/stm32/newlib.cpp causes linker errors
+void *__dso_handle = NULL;
```

Comment:
Do you know what has changed to make this necessary?

Reply:
`part1` now has a global variable (`Diagnostics`) with a non-trivial destructor